mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp

```diff
@@ -207,5 +208,55 @@ void SYCLConstructorOp::getEffects(
   }
 }
 
+void SYCLIDConstructorOp::getEffects(
+    SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
+        &effects) {
+  auto *defaultResource = SideEffects::DefaultResource::get();
+  // All of the arguments will be scalar or read from
+  for (auto value : getArgs()) {
+    if (isa<MemRefType, LLVM::LLVMPointerType>(value.getType())) {
+      effects.emplace_back(MemoryEffects::Read::get(), value, defaultResource);
+    }
+  }
+  // The result will be allocated
+  effects.emplace_back(MemoryEffects::Allocate::get(), getId(),
```

Comment:
Wondering whether the output argument (the SYCL id being constructed) should also have the MemoryEffects::Write, IT is allocated but also initialized so I'd say it should have it.

Reply:
Yes, that would not hurt. Added.